itest/psbt_test.go

```diff
@@ -0,0 +1,336 @@
+
+	// Now we create a script tree consisting of two simple scripts.
+	preImage := []byte("hash locks are cool")
+	leaf1 := test.ScriptHashLock(t.t, preImage)
+	leaf2 := test.ScriptSchnorrSig(t.t, bobScriptKey.PubKey)
+	leaf1Hash := leaf1.TapHash()
+	leaf2Hash := leaf2.TapHash()
+	tapscript := input.TapscriptPartialReveal(
+		bobScriptKey.PubKey, leaf2, leaf1Hash[:],
+	)
+	rootHash := tapscript.ControlBlock.RootHash(leaf2.Script)
+
+	controlBlockBytes, err := tapscript.ControlBlock.ToBytes()
+	require.NoError(t.t, err)
+
+	bobAssetScriptKey, err := tapscript.TaprootKey()
+	require.NoError(t.t, err)
+
+	t.Logf("Bob destination key %x (internal %x, root %x)",
+		bobAssetScriptKey.SerializeCompressed(),
+		bobScriptKey.PubKey.SerializeCompressed(), rootHash[:])
+
+	// Next, we'll attempt to complete a transfer with PSBTs from our main
+	// node to Bob.
+	const numUnits = 10
+	bobAddr, err := bob.NewAddr(ctxb, &tarorpc.NewAddrRequest{
+		GenesisBootstrapInfo: genBootstrap,
+		Amt:                  numUnits,
+		ScriptKey: &tarorpc.ScriptKey{
+			PubKey:   schnorr.SerializePubKey(bobAssetScriptKey),
+			KeyDesc:  lndKeyDescToTaro(bobScriptKey),
+			TapTweak: rootHash[:],
+		},
+		InternalKey: lndKeyDescToTaro(bobInternalKey),
+	})
+	require.NoError(t.t, err)
+	assertAddrCreated(t.t, bob, rpcAssets[0], bobAddr)
+
+	// Send the asset to Bob using the script key with an actual script
+	// tree.
+	sendResp := sendAssetsToAddr(t, alice, bobAddr)
+
+	changeUnits := rpcAssets[0].Amount - numUnits
+	confirmAndAssertOutboundTransfer(
+		t, alice, sendResp, genInfo.AssetId, changeUnits, 0, 1,
+	)
+	_ = sendProof(t, alice, bob, bobAddr, genInfo)
+	assertReceiveComplete(t, bob, 1)
+
+	// Now try to send back those assets using the PSBT flow.
+	aliceAddr, err := alice.NewAddr(ctxb, &tarorpc.NewAddrRequest{
+		GenesisBootstrapInfo: genBootstrap,
+		Amt:                  numUnits / 2,
+	})
+	require.NoError(t.t, err)
+	assertAddrCreated(t.t, alice, rpcAssets[0], aliceAddr)
+
+	fundResp := fundAddressSendPacket(t, bob, aliceAddr)
+	t.Logf("Funded PSBT: %v",
+		base64.StdEncoding.EncodeToString(fundResp.FundedPsbt))
+
+	fundedPacket, err := taropsbt.NewFromRawBytes(
+		bytes.NewReader(fundResp.FundedPsbt), false,
+	)
+	require.NoError(t.t, err)
+
+	// We can now ask the wallet to sign the script path, since we only need
+	// a signature.
+	fundedPacket.Inputs[0].TaprootMerkleRoot = rootHash[:]
+	fundedPacket.Inputs[0].TaprootLeafScript = []*psbt.TaprootTapLeafScript{
+		{
+			ControlBlock: controlBlockBytes,
+			Script:       leaf2.Script,
+			LeafVersion:  leaf2.LeafVersion,
+		},
+	}
+	fundedPacket.Inputs[0].TaprootBip32Derivation[0].LeafHashes = [][]byte{
+		leaf2Hash[:],
+	}
+	var b bytes.Buffer
+	err = fundedPacket.Serialize(&b)
+	require.NoError(t.t, err)
+
+	signedResp, err := bob.SignVirtualPsbt(
+		ctxb, &wrpc.SignVirtualPsbtRequest{
+			FundedPsbt: b.Bytes(),
+		},
+	)
+	require.NoError(t.t, err)
+	require.Contains(t.t, signedResp.SignedInputs, uint32(0))
```

Comment:
Hmm, but since `SignVirtualPsbt` explicitly mentions virtual transactions in its name, I think it should be clear what the indexes refer to? Added a comment anyway.